mllib/src/main/scala/org/apache/spark/ml/regression/DecisionTreeRegressor.scala

```diff
@@ -113,6 +114,17 @@ final class DecisionTreeRegressionModel private[ml] (
     rootNode.predictImpl(features).prediction
   }
 
+  def predictVariance(features: Vector): Double = {
```

Comment:
make protected for now.

Also, add a note that this will require updating if we ever add other impurity measures.